google/api_version_converters.go

```diff
@@ -0,0 +1,80 @@
+package google
+
+import (
+	"encoding/json"
+	"google.golang.org/api/compute/v1"
+)
+
+func convert(item, out interface{}) error {
+	bytes, err := json.Marshal(item)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(bytes, out)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func convertInstanceGroupManagerToV1(self interface{}) (*compute.InstanceGroupManager, error) {
```

Comment:
We could probably eliminate this - I like hiding the empty structs you need to instantiate from the caller, but it does add a lot error handling LOC.

Another thing we could consider is upgrading the errors here to panics - they are unrecoverable for the caller, since we are really just using json to do a cast; fixing it would require a recompile. Then, the return would only be the new struct, and we would have a much cleaner interface for the caller.